src/services/key/KeyService.php

```diff
@@ -73,6 +73,6 @@ private function downloadKey(string $keyId): PublicKey {
     }
 
     private function allowedToImport(PublicKey $key): bool {
-        return $this->trusted->has($key->getId()) || $this->input->confirm('Import this key?', false);
+        return $this->input->isNoInteraction() || $this->trusted->has($key->getId()) || $this->input->confirm('Import this key?', false);
```

Comment:
Is that logically correct?

As I read it, this would auto-accept an unknown key in _no-interaction_-mode. That's the reverse of the default behavior of the `confirm` call. And I'm not convinced we want the default behavior for _no-interaction_ to auto accept?

Reply:
Isn't that even redundant? Given that `ConsoleInput::confirm` will also honor the _no-interaction_ flag?

Reply:
@theseer 

👍

Reply:
@theseer 

Ha, actually, no - because when the `--no-interaction` option is used, `$input->confirm($question, $default)` returns the `$default`, which is `false` here!

Reply:
So just to be 100% sure: It's the intention of the `no-interaction`-flag to auto confirm everything? Even import keys and such? How about conflicting / changed keys?

I'm not sure I like that - but maybe I'm just paranoid.
How do other tools/platforms do that?

Reply:
@localheinz ping?

Reply:
@theseer 

Hmm, yes - to avoid asking interactive questions in continuous integration environments, for example!

What do you think?

Reply:
I'm undecided.

I do understand that none interactive things make sense. What I'm not sure about is whether that should imply "yes" to everything - even key related things.

So I was wondering how other tools deal with that?

Reply:
I know that `dnf` has an `--assumeyes` switch... Not sure if that applies to key imports as well.

Reply:
@theseer 

How do you feel about renaming the option (and possibly limiting the effect of its usage) to clarify what it does?

Reply:
I have to admit that I like `dnf`'s `--assumeyes`. With that name, I probably even would except that keys will automagically get accepted.

Phive though remembers the key used for previous installs, warning when the key changed. I feel that would be something to NOT auto-accept but fail.

What do you think?

Reply:
> Phive though remembers the key used for previous installs, warning when the key changed. I feel that would be something to NOT auto-accept but fail.

I like that. (My usecase for this feature is running `phive install` in CI builds, and there the keys are not stored between builds.)

Reply:
I do understand that use case, but I think `--assumeyes` or whatever we call it is the wrong solution.

I guess we agree that the sole reason to check a signature before installing is to ensure the file has not been messed with after it was signed using a trusted key.

Now, if you auto-import any "random" key that just happens to have been used, that renders the whole process useless and you could just as well not check the signature.

So instead of generically accepting any random key, we opted to implement a means to provide a list of "trusted" key ids (`--trust-gpg-keys`).

If an unknown key is used and its ID is trusted, we auto import it.

I do see that's not the same, but the only secure approach as far as I can see.

Reply:
@theseer With `---trust-gpg-keys` and the key auto-import, are you referring to the current functionality of the released PHIVE version or to the changes in the PR?

Reply:
That is existing behavior.

Reply:
```
theseer@nyda ~ $ phive --help | grep trust
        --trust-gpg-keys         Silently import these keys when required (separated by comma)
```

Reply:
In my experience, the "silently" part does not work: PHIVE still asks whether it should import these keys. (This only happens with `phive install`, not with `phive update packagename`.)

Reply:
That would be a bug. I wouldn't know how that could happen though.

I know there is an edge case when a key changed, even if you explicitly trust the new key, it will still complain and not accept it. I yet have to figure out why that is happening.